qiskit/visualization/state_visualization.py

```diff
@@ -662,6 +663,8 @@ def plot_state_qsphere(state, figsize=None, ax=None, show_state_labels=True,
             show the phase for each basis state.
         use_degrees (bool): An optional boolean indicating whether to use
             radians or degrees for the phase values in the plot.
+        show_all_state_labels (bool): An optional boolean indicating whether to
+            show labels for all basis states.
```

Comment:
Thanks @mtreinish for reviewing this PR. Yes, you are absolutely right, this argument would clutter the image for more than 3 qubits. That is why I marked it 'False' by default. Only motive of introducing this new argument is to let the interested user know visually where on the qsphere are all the states located and which ones out of all possible states have magnitude. 2D view is cluttered but may be a 3D view would carry this argument well without clutter.  

I will make the changes in documentation as per your suggestion, so that the user is well aware of what he might encounter.